src/go/k8s/pkg/resources/certmanager/type_helpers.go

```diff
@@ -245,6 +251,7 @@ func (cc *ClusterCertificates) prepareAPI(
 	// every time both listeners share the same set of certificates
 	nodeSecretRef := tlsListeners[0].GetTLS().NodeSecretRef
 	result.externalNodeCertificate = nodeSecretRef
+	result.selfSignedNodeCertificate = tlsListeners[0].GetTLS().IssuerRef == nil && nodeSecretRef == nil
```

Comment:
yes you're right - but this already covers many use cases. I think extending it to self-signed injected would add another couple of days of work  because I think the only way is to kubectl get the certificate and find out probably (?)

Reply:
fixed this